lib/trie/database.go

```diff
@@ -363,13 +374,15 @@ func (t *Trie) writeDirty(db chaindb.Batch, n Node) error {
 			hash, err)
 	}
 
-	branch, ok := n.(*node.Branch)
-	if !ok {
-		// the node is a leaf
+	switch n.Type() {
+	case node.BranchType, node.BranchWithValueType:
+	default: // not a branch
```

Comment:
```suggestion
	default: // not a branch, so is a leaf
```

lol, very minor, up to you to change this

Reply:
Oops sorry I didn't see your comment, my bad!
It could be an untyped nil as well eventually :thinking: (not super sure, my 2021 memory faded a bit)